### PR TITLE
fix(thorvg): build static library without -fPIC

### PR DESCRIPTION
**Pull request by suda-morris**
ESP-IDF only use static library, and the thorvg library won't be linked into a dynamic library.

This PR also enables the multi-threading feature in thorvg.

```
Summary:
    ThorVG version:             0.13.6
    Build Type:                 debugoptimized
    Prefix:                     /usr/local
    Multi-Tasking:              true                        <------------- Enabled in this PR
    SIMD Instruction:           none
    Raster Engine (SW):         true
    Raster Engine (GL_BETA):    false
    Raster Engine (WG_BETA):    false
    Loader (TVG):               true
    Loader (SVG):               true
    Loader (TTF):               false
    Loader (LOTTIE):            true
    Loader (PNG):               false
    Loader (JPG):               false
    Loader (WEBP):              false
    Saver (TVG):                false
    Saver (GIF):                false
    Binding (CAPI):             true
    Binding (WASM_BETA):        false
    Log Message:                false
    Tests:                      false
    Examples:                   false
    Tool (Svg2Tvg):             false
    Tool (Svg2Png):             false
    Tool (Lottie2Gif):          false
    Extra (Lottie Expressions): true
```

### CODE REVIEW

File: thorvg/CMakeLists.txt [hunk 1]
```diff
@@ -94,14 +94,15 @@ ExternalProject_Add(${TVG_LIB}_target
...
+    -Ddefault_library=static # build static library
+    -Db_staticpic=false # no -fPIC
+    -Dthreads=true # allow multi-threading
     BUILD_COMMAND ninja -C ${TVG_BUILD_DIR}
     INSTALL_COMMAND ""
     BUILD_BYPRODUCTS ${TVG_BUILD_DIR}/src/${TVG_LIB}.a
 )
 
-add_prebuilt_library(${TVG_LIB} ${TVG_BUILD_DIR}/src/${TVG_LIB}.a PRIV_REQUIRES esp_rom)
+add_prebuilt_library(${TVG_LIB} ${TVG_BUILD_DIR}/src/${TVG_LIB}.a PRIV_REQUIRES pthread)
```
Comment (igrr) [inline thorvg/CMakeLists.txt:105]: Is longjmp no longer required after enabling pthreads?

File: thorvg/CMakeLists.txt [hunk 2]
```diff
@@ -80,7 +80,7 @@ endif()
 
 file(APPEND ${TVG_CROSS_CFG} "
 [host_machine]
-system = 'freertos'
+system = 'android'
```
Comment (igrr) [inline thorvg/CMakeLists.txt:83]: I recall you asked the question about this in the previous PR... Did you figure out why this has to be `android`?

**Review by igrr — APPROVED** [review]

File: thorvg/CMakeLists.txt (hunk 1 above)
Comment (suda-morris) [reply thorvg/CMakeLists.txt:105]: The `__wrap_longjmp` is still needed, but pthread may pull the esp_rom dependency implicitly.

File: thorvg/CMakeLists.txt (hunk 2 above)
Comment (suda-morris) [reply thorvg/CMakeLists.txt:83]:
```meson
if get_option('threads') == true and host_machine.system() != 'windows' and host_machine.system() != 'android'
    thread_dep = meson.get_compiler('cpp').find_library('pthread')
    thorvg_lib_dep += [thread_dep]
endif
```

the meson build system can't find the pthread library in the default search path.  

I tried added `cpp_link_args = ['-L${CMAKE_BINARY_DIR}/esp-idf/pthread', '-lpthread']` to the mseon cross file, still no luck.

**Pull request merged**